src/metabase/lib/js.cljs

```diff
@@ -843,80 +1339,130 @@
       (to-array (lib.core/expressionable-columns a-query stage-number expression-position)))))
 
 (defn ^:export suggested-join-conditions
-  "Return suggested default join conditions when constructing a join against `joinable`, e.g. a Table, Saved
-  Question, or another query. Suggested conditions will be returned if the source Table has a foreign key to the
-  primary key of the thing we're joining (see #31175 for more info); otherwise this will return `nil` if no default
-  conditions are suggested."
+  "Returns a JS array of possible default join conditions when joining against `joinable`, e.g. a Table, Saved
+  Question, or another query. Suggested conditions will be returned if the existing query has a foreign key to the
+  primary key of the `joinable`. (See #31175 for more info.)
+
+  Returns `[]` if we cannot determine any \"obvious\" join conditions.
+
+  > **Code health:** Healthy"
   [a-query stage-number joinable]
   (to-array (lib.core/suggested-join-conditions a-query stage-number joinable)))
 
 (defn ^:export join-fields
-  "Get the `:fields` associated with a join."
+  "Get the fields list associated with `a-join`. That is, the set of fields from the *joinable* which are being joined
+  into the query.
+
+  This is either a JS array of columns, or one of the keywords `:all` or `:none`.
+
+  > **Code health:** Smelly. It's fine to call this, but it's one of the few places that returns field refs directly.
```

Comment:
I think this is the same as `fields` and `with-fields`. Returns opaque clauses (which happen to be refs).